### PR TITLE
Rover: calc_steering_from_turn_rate remove unused arguments

### PR DESCRIPTION
**Pull request by IamPete1**
Was poking about looking at https://github.com/ArduPilot/ardupilot/pull/15442 and spotted this.

Quite surprised we didn't get a compiler error for this one!

### CODE REVIEW

File: Rover/mode.h [hunk 1]
```diff
@@ -165,7 +165,7 @@ class Mode
     void navigate_to_waypoint();
 
     // calculate steering output given a turn rate and speed
```
Comment (peterbarker) [inline Rover/mode.h:167]: You missed a bit

```suggestion
    // calculate steering output given a turn rate
```

**Review by peterbarker — COMMENTED** [review]
Would be nice to have a comment on what the units of `turn_rate` here are.

**Comment by rmackay9** [comment]
@IamPete1,

This looks good to me but I guess the DO_REVERSED mission command still works ok?  Clearly this PR shouldn't break it because it's apparently a non-functional change but it would be good to double check.  thanks!

File: Rover/mode.h (hunk 1 above)
Comment (IamPete1) [reply Rover/mode.h]: fixed and added a units comment

**Comment by IamPete1** [comment]
@rmackay9 yep DO_REVERSED still works, just tested in SITL

**Review by rmackay9 — APPROVED** [review]
Great, let's merge after it passes travis, thanks!

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**